### PR TITLE
Initial procedure to migrate large data sets to hypertable

### PR DESCRIPTION
**Pull request by davidkohn88**
Co-authored-by: Ryan Booz <ryan@ryanbooz.com>

### CODE REVIEW

**Comment by davidkohn88** [comment]
Fixes #19

**Comment by jonatas** [comment]
This is very cool! I remember the first time I was testing timescale this was the first thing I did. Try to [hyper_move](https://gist.github.com/jonatas/d040a216ea39ae774ef11e87da74d21c) all the columns, but not in batches!

Maybe it can become a function available in timescaledb boosting the extra parameters of `create_hypertable`:

```sql
SELECT create_hypertable('sensor_data', 'time', 
  chunk_time_interval=>'1 hour'::interval, 
  'migrating_from' => 'sensor_data_copy',
  in_batches => 10000)
```

File: migrate.sql [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    dimension_row _timescaledb_catalog.dimension;
+
+    min_time_internal bigint;
+    max_time_internal bigint;
+    interval_internal bigint;
+    sql_text text;
+BEGIN
+    SELECT (get_dimension_details(sink_table)).* INTO STRICT dimension_row;
+    
+    --Get the min and max times in timescale internal format from the source table, this will tell us which chunks we need to decompress
```
Comment (JLockerman) [inline migrate.sql:59]: comment out of date?

File: migrate.sql [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+   -- populate the table with rows for each chunk
+   -- of data to process
+   SELECT dimension_row.interval_length/10 into interval_internal;
+
+
+EXECUTE FORMAT($$
+    INSERT INTO %1$I (start_t, end_t, parallel_worker_num, migrated) 
+    SELECT s as start_t, s + %4$s as end_t, %5$s::INT as parallel_worker_num, false 
+FROM (select generate_series(%2$s, %3$s,%4$s) as s) f ON CONFLICT DO NOTHING $$, 
+log_table_name, min_time_internal,max_time_internal, interval_internal,parallel_worker_num);
```
Comment (JLockerman) [inline migrate.sql:89]: indentation

**Pull request merged**